pkg/stores/proxy/proxy_store.go

```diff
@@ -327,6 +342,12 @@ func (s *Store) listAndWatch(apiOp *types.APIRequest, client dynamic.ResourceInt
 	eg.Go(func() error {
 		for event := range watcher.ResultChan() {
 			if event.Type == watch.Error {
+				if status, ok := event.Object.(*metav1.Status); ok {
+					logrus.Debugf("event watch error: %s", status.Message)
+					returnErr(errors.Wrapf(err, "event watch error: %+v", event), result)
```

Comment:
There may be a situation where `err` is `nil`. If `err` is `nil` we don't get any useful error msg from the UI(websocket). It is best to add an `if` statement. When `err` is `nil`, we should use `errors.Errorf`.

Reply:
In addition, we need to pay attention that  [rancher/apiserver](https://github.com/rancher/apiserver/blob/master/pkg/subscribe/watcher.go#L30)) will send `resource.stop` event notifications, combined with the current logic of [rancher/dashboard resource.error](https://github.com/rancher/dashboard/blob/master/shell/plugins/steve/subscribe.js#L437) & [rancher/dashboard resource.stop](https://github.com/rancher/dashboard/blob/master/shell/plugins/steve/subscribe.js#L460), possibly sending `resource.error` event will have other effects.